### PR TITLE
SIE-76 face detection with bucket

### PR DESCRIPTION
**Pull request by jaynilpatel**
### Changes
* Added `trigger_detect_and_mask` function. Triggers when a file is added to `sie-raw-images` bucket. 
* Once the mask is created, it is uploaded to `sie-masked-images` bucket
### Testing
To test functionality, go to a `sie-raw-images` bucket and upload an image `participant_id.jpg`. A masked image will be added to `sie-masked-images` at `participant_id/neutral.jpg` in few seconds. Open cloud function logs to see errors if any.

### CODE REVIEW

File: image_processing/constants.py [hunk 1]
```diff
@@ -22,6 +22,6 @@
 """
     PROJECT SPECIFIC
 """
-TEMP_DIR = "temp"
+TEMP_DIR = "tmp"
```
Comment (quanganhhoang) [inline image_processing/constants.py]: I think `/tmp` might be better here

File: image_processing/main.py [hunk 2]
```diff
@@ -196,11 +198,8 @@ def process_img(
     final_img = resize(gray_img)
 
     # Save processed image to local dir
-    processed_file_path = (
-        os.getcwd() + f"/{constants.TEMP_DIR}/{constants.PROCESSED_IMAGE}"
-    )
+    processed_file_path = f"/{constants.TEMP_DIR}/{constants.PROCESSED_IMAGE}"
```
Comment (quanganhhoang) [inline image_processing/main.py]: then remove the first `/` here

File: image_processing/main.py [hunk 3]
```diff
@@ -411,3 +410,102 @@ def upload_processed_images(bucket_name, source_file_folder):
...
+
+    bucket_name = event["bucket"]  # sie-raw-images bucket
+    cloud_storage_prefix = "gs://" + bucket_name + "/"
+    cloud_download_from = event["name"]  # file path
+
+    # extract file name from path eg: folder/image.jpg
+    file_name = cloud_download_from.split("/")[-1]
+
+    # local directory to store image
+    cloud_download_to = "/" + constants.TEMP_DIR + "/" + file_name
```
Comment (quanganhhoang) [inline image_processing/main.py]: remove first `"/"`. Or you can do this `f"{constants.TEMP_DIR}/{file_name}"`

File: image_processing/main.py [hunk 4]
```diff
@@ -411,3 +410,102 @@ def upload_processed_images(bucket_name, source_file_folder):
...
+        )
+
+        # Upload masked image to sie-masked-images bucket
+        participant_id = os.path.splitext(event["name"])[0]  # remove file ext
+        cloud_upload_to = f"{participant_id}/{constants.PROCESSED_IMAGE}"
+        upload_image("sie-masked-images", cloud_upload_from, cloud_upload_to)
+        print("Processed image saved at: " + cloud_upload_to)
+
+    except exceptions.InvalidFaceImage as err:
+        traceback.print_exception(*sys.exc_info())
```
Comment (quanganhhoang) [inline image_processing/main.py:505]: i'm curious here ~ why did you use `traceback.print_exception(*sys.exc_info())`?

**Review by quanganhhoang — CHANGES_REQUESTED** [review]
LGTM! Just a few nit comments. Btw, since we won't need to store the images in `sie-raw-images`, maybe we can have a separate PR to clean up user-uploaded image after we're done with our image-processing. What do you think?

File: image_processing/constants.py (hunk 1 above)
Comment (Abinashi7) [reply image_processing/constants.py]: @quanganhhoang  I think that's a great idea to create a separate PR to clean up the images after they are processed.

**Review by Abinashi7 — APPROVED** [review]

File: image_processing/main.py (hunk 4 above)
Comment (jaynilpatel) [reply image_processing/main.py:505]: Without this, there were no error messages printing in the logs. This line will print a comprehensive stack trace which helps in debugging.

Comment (quanganhhoang) [reply image_processing/main.py:505]: TIL! Thanks!

**Review by quanganhhoang — APPROVED** [review]

**Pull request merged**